node-api/handlers/beacon/validators.go

```diff
@@ -22,45 +22,36 @@ package beacon
 
 import (
 	"errors"
+	"fmt"
 	"net/http"
 
 	"github.com/berachain/beacon-kit/node-api/backend"
 	"github.com/berachain/beacon-kit/node-api/handlers"
 	beacontypes "github.com/berachain/beacon-kit/node-api/handlers/beacon/types"
 	types "github.com/berachain/beacon-kit/node-api/handlers/types"
 	"github.com/berachain/beacon-kit/node-api/handlers/utils"
+	"github.com/berachain/beacon-kit/primitives/math"
 )
 
 var ErrNoSlotForStateRoot = errors.New("slot not found at state root")
 
-// getStateValidators is a helper function to provide implementation
-// consistency between GetStateValidators and PostStateValidators, since they
-// are intended to behave the same way.
-func (h *Handler) getStateValidators(stateID string, ids []string, statuses []string) (any, error) {
```

Comment:
moved below and renamed `FilterValidators`.
Exported and without any reference to http request/response, it exposes validators filtering logic which can be reused in simulated tests